### PR TITLE
Allow to specify timeout values using ENV vars

### PR DESCRIPTION
**Pull request by clalbus**
In our environment default 5s timeout values were too low. This would allow everyone to fine tune them for specific use cases.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,7 +13,13 @@ ENV NAMESERVERS="208.67.222.222 8.8.8.8 208.67.220.220 8.8.4.4" \
...
     TALK=talk:100 \
     UDP=0 \
-    VERBOSE=0
+    VERBOSE=0 \
+    TIMEOUT_CLIENT=5s \
+    TIMEOUT_CLIENT_FIN=5s \
+    TIMEOUT_CONNECT=5s \
+    TIMEOUT_SERVER=5s \
+    TIMEOUT_SERVER_FIN=5s \
+    TIMEOUT_TUNNEL=5s
```
Comment (yajo) [inline Dockerfile]: Please sort them alphabetically.

File: magic-entrypoint.py [hunk 2]
```diff
@@ -9,8 +9,15 @@
...
-LISTENS = os.environ["LISTEN"].split()
-NAMESERVERS = os.environ["NAMESERVERS"].split()
+LISTENS            = os.environ["LISTEN"].split()
+NAMESERVERS        = os.environ["NAMESERVERS"].split()
+TIMEOUT_CLIENT     = os.environ["TIMEOUT_CLIENT"]
+TIMEOUT_CLIENT_FIN = os.environ["TIMEOUT_CLIENT_FIN"]
+TIMEOUT_CONNECT    = os.environ["TIMEOUT_CONNECT"]
+TIMEOUT_SERVER     = os.environ["TIMEOUT_SERVER"]
+TIMEOUT_SERVER_FIN = os.environ["TIMEOUT_SERVER_FIN"]
+TIMEOUT_TUNNEL     = os.environ["TIMEOUT_TUNNEL"]
```
Comment (yajo) [inline magic-entrypoint.py]: Please remove this diff. (See below)

File: magic-entrypoint.py [hunk 3]
```diff
@@ -22,19 +29,19 @@
...
-    timeout connect 5s
-    timeout server 5s
-    timeout server-fin 5s
-    timeout tunnel 5s
+    timeout client {TIMEOUT_CLIENT}
+    timeout client-fin {TIMEOUT_CLIENT_FIN}
+    timeout connect {TIMEOUT_CONNECT}
+    timeout server {TIMEOUT_SERVER}
+    timeout server-fin {TIMEOUT_SERVER_FIN}
+    timeout tunnel {TIMEOUT_TUNNEL}
```
Comment (yajo) [inline magic-entrypoint.py]: Use `env(TIMEOUT_TUNNEL)` and similars, which is [supported natively by HAProxy](http://cbonte.github.io/haproxy-dconv/1.9/configuration.html#7.3.2-env).

File: magic-entrypoint.py [hunk 4]
```diff
@@ -22,19 +29,19 @@
     bind {listen}
     default_backend talk_{index}
 """
-config = """
+config = f"""
```
Comment (yajo) [inline magic-entrypoint.py]: Remove please (see below)

**Review by yajo — CHANGES_REQUESTED** [review]
Overall it looks pretty good.

I just added some comments on code style and specific implementation of variable interpolation. Please fix them to merge.

Thanks for your contribution! :clap: :clap:

**Comment by clalbus** [comment]
Thank you for the comments, I have fixed all the points you have raised.

File: magic-entrypoint.py [hunk 5]
```diff
@@ -9,8 +9,9 @@
 
 logging.root.setLevel(logging.INFO)
 
-LISTENS = os.environ["LISTEN"].split()
-NAMESERVERS = os.environ["NAMESERVERS"].split()
+LISTENS            = os.environ["LISTEN"].split()
+NAMESERVERS        = os.environ["NAMESERVERS"].split()
+
```
Comment (yajo) [inline magic-entrypoint.py]: Please could you revert this diff? It's not needed anymore

Comment (clalbus) [reply magic-entrypoint.py]: Fixed, sorry about that.

**Review by yajo — APPROVED** [review]

**Pull request merged**